### PR TITLE
Added an ignore_for_file option to ignore linter rules for generated files (Fix #110)

### LINKED ISSUE #110: Ignore analyzer rules on generated files
Hi,

Generated files are not following some analyzer rules (ex: public_member_api_docs), would it be possible to add an `//ignore_for_file: ..` with rules ?

Or even better, as possible with json_serializable (coming from source_gen: https://pub.dev/packages/source_gen#configuring-combining_builder-ignore_for_file) a way to specify in bulid.yaml; some rules to ignore in generated files.

```
source_gen|combining_builder:
        options:
          ignore_for_file:
            - public_member_api_docs
```

Thanks :)

### PR DESCRIPTION
**Pull request by MattisBrizard**
Hi, I created this PR to fix the issue I opened. #110 
Tell me if I missed something.

Mattis

### CODE REVIEW

**Comment by vasilich6107** [comment]
Hi! Thanks for PR.
Could you rebase on latest `beta` and update the versions in `pubspec` and `changelog`

**Comment by MattisBrizard** [comment]
Hi @vasilich6107, done.
Waiting for the review.

**Comment by vasilich6107** [comment]
LGTM - let's as @comigor

**Comment by comigor** [comment]
I've just rebased to make it easier to review/merge, sorry for taking long

File: README.md [hunk 1]
```diff
@@ -126,6 +126,7 @@ Each `SchemaMap` is configured this way:
 | `queries_glob` |  | Glob that selects all query files to be used with this schema. |
 | `naming_scheme` | `pathedWithTypes` | The naming scheme to be used on generated classes names. `pathedWithTypes` is the default for retrocompatibility, where the names of previous types are used as prefix of the next class. This can generate duplication on certain schemas. With `pathedWithFields`, the names of previous fields are used as prefix of the next class and with `simple`, only the actual GraphQL class nameis considered. | 
 | `type_name_field` | `__typename` | The name of the field used to differentiatiate interfaces and union types (commonly `__typename` or `__resolveType`). Note that `__typename` field are not added automatically to the query. If you want interface/union type resolution, you need to manually add it there. |
+| `ignore_for_file` |  | The linter rules to ignore for artemis generated files. |
```
Comment (comigor) [inline README.md:129]:
```suggestion
| `ignore_for_file` | `[]`  | The linter rules to ignore for artemis generated files. |
```

**Review by comigor — APPROVED** [review]
Thank you very much for the PR!

**Pull request merged**